snap_plugin/v1/plugin.py

```diff
@@ -439,6 +487,35 @@ def _parse_args(self):
                   self._is_shutting_down),
             kwargs={"timeout": self._get_ping_timeout_duration()})
 
+    def _tls_setup(self):
+        # check if the secure flag has been added
+        self.meta.server_cert_path = self._args.cert_path
+        self.meta.private_key_path = self._args.key_path
+        self.meta.root_cert_paths = self._args.root_cert_paths.split(":") if self._args.root_cert_paths is not None else None 
+        # Check that all required variables are set
+        if self.meta.private_key_path is None:
+            raise MissingRequiredArgument("key-path", "tls")
+        elif self.meta.server_cert_path is None:
+            raise MissingRequiredArgument("cert-path", "tls")
+        elif self.meta.root_cert_paths is None:
+            raise MissingRequiredArgument("root-cert-paths", "tls")
+        if self.meta.cipher_suites is not None:
+            os.environ["GRPC_SSL_CIPHER_SUITES"] = ":".join(self.meta.cipher_suites)
+
+    def _generate_tls_credentials(self):
+        blocks = []
+        for path in self.meta.root_cert_paths:
+            try:
+                cert = open(path).read().encode()
```

Comment:
Actually elements of `root_cert_paths` may also be directories. Directories found on the list should be scanned for certificate files (but no deeper).
See implementation in [plugin-lib-go](https://github.com/intelsdi-x/snap-plugin-lib-go/blob/cdca5a915f3fb0f34095d2c6d85eebda4313c24f/v1/plugin/plugin.go#L238).